### PR TITLE
[v0.31] Price-type functionality proper

### PR DESCRIPTION
**Pull request by petioptrv**
**Before submitting this PR, please make sure**:
- [X] Your code builds clean without any errors or warnings
- [X] You are using approved title ("feat/", "fix/", "docs/", "refactor/", etc)

**Optional**:
- Related Github issue: #1919
- Related Clubhouse Story:

**A description of the changes proposed in the pull request**:

### CODE REVIEW

**Comment by Nullably** [comment]
@petioptrv I have just updated the issue
https://github.com/CoinAlpha/hummingbot/issues/1919#issuecomment-667763312
Sorry, I should have done this sooner. 
Let me know what you think.

The code looks good to me, thanks.

File: hummingbot/strategy/pure_market_making/pure_market_making_config_map.py [hunk 1]
```diff
@@ -288,28 +296,38 @@ def exchange_on_validated(value: str):
                   default="current_market",
                   validator=validate_price_source,
                   on_validated=on_validate_price_source),
+    "price_type":
+        ConfigVar(key="price_type",
+                  prompt="Which price type to use? (mid_price/last_price) >>> ",
```
Comment (Nullably) [inline hummingbot/strategy/pure_market_making/pure_market_making_config_map.py:301]: Options here should be mid_price/last_price/best_bid/best_ask (as many as PriceType enum has).

**Comment by mifeng** [comment]
@petioptrv Apologies for the delay in getting this fix merged. Our team spent a lot of time during v0.30 release refactoring and cleaning up the codebase, as well as adding the `last_trade` endpoints that were needed for this.

Since we already code freezed v0.30, we will aim to get this merged into the v0.31 release. The code freeze date for v0.31 will be in early September, but we'll try to get this merged into `development` even sooner.

File: hummingbot/strategy/pure_market_making/pure_market_making_config_map.py (hunk 1 above)
Comment (petioptrv) [reply hummingbot/strategy/pure_market_making/pure_market_making_config_map.py:301]: I started making the change, but I think that `AssetPriceDelegate` only offers "mid_price". How should we proceed in that case? External market always = mid-price for now?

Comment (Nullably) [reply hummingbot/strategy/pure_market_making/pure_market_making_config_map.py]: These new four price types should apply to external market as well. 
Can you please modify the `AssetPriceDelegate`? Thanks

**Comment by mifeng** [comment]
@petioptrv I'd to ask you about something unrelated to this PR. 

Can you send an email to me mike[at]hummingbot.io? Thanks!

**Comment by petioptrv** [comment]
@Nullably, I still have to make the PMM test case pass, but I wanted to get your input on the asset price delegate test case. Is there a way to simulate a trade on the external market? If not, I can just keep the first assertion, which already passes.

**Comment by Nullably** [comment]
@petioptrv 
I think what you did for the changes and testing is good enough. And from the code, looks like you already simulated the trade on the external market.

**Comment by petioptrv** [comment]
> @petioptrv
> I think what you did for the changes and testing is good enough. And from the code, looks like you already simulated the trade on the external market.

Hey @Nullably, my mistake; I should have marked the PR as a draft... The test cases were not passing. I attempted to simulate the trade on the external market, but that didn't work properly. I left the line there to illustrate the intent.

**Comment by petioptrv** [comment]
I've pushed the changes necessary to make the tests pass.

**Comment by Nullably** [comment]
@petioptrv Sorry, where did you push the changes to? 
Can you make another PR to `price-source` feature branch?
Thanks

**Comment by petioptrv** [comment]
> @petioptrv Sorry, where did you push the changes to?
> Do you want me to revert the changes or you want to make another PR?
> Thanks

I pushed the [changes](https://github.com/petioptrv/hummingbot/commit/3c69bf1f91b32f147f210c914cfa783e99b81198) to the PR [branch on my fork](https://github.com/petioptrv/hummingbot/commits/feat/price-source-type-proper). Not sure how to handle this. Can we re-open the PR to accept those changes?

**Comment by Nullably** [comment]
I reverted the merge. Not sure if you can reopen this PR, otherwise you can create a new one.

**Pull request merged**